src/main/java/org/radarbase/appserver/service/protocol/GithubClient.java

```diff
@@ -0,0 +1,93 @@
+/*
+ *
+ *  *
+ *  *  * Copyright 2018 King's College London
+ *  *  *
+ *  *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  *  * you may not use this file except in compliance with the License.
+ *  *  * You may obtain a copy of the License at
+ *  *  *
+ *  *  *   http://www.apache.org/licenses/LICENSE-2.0
+ *  *  *
+ *  *  * Unless required by applicable law or agreed to in writing, software
+ *  *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  *  * See the License for the specific language governing permissions and
+ *  *  * limitations under the License.
+ *  *  *
+ *  *
+ *
+ */
+
+package org.radarbase.appserver.service.protocol;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+
+@Slf4j
+@Component
+@Scope(value = ConfigurableBeanFactory.SCOPE_SINGLETON)
+public class GithubClient {
+
+    private static final String GITHUB_API_URI = "api.github.com";
+    private static final String GITHUB_API_ACCEPT_HEADER = "application/vnd.github.v3+json";
+    private final transient ObjectMapper objectMapper;
+    private final transient HttpClient client;
+
+    @Value("${security.github.client.token}")
+    private transient String githubToken;
+
+    @SneakyThrows
+    @Autowired
+    public GithubClient(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+        client = HttpClient.newBuilder().connectTimeout(Duration.ofSeconds(10)).build();
```

Comment:
probably best to specify a read time out too

Reply:
Would that be here already: 
https://github.com/RADAR-base/RADAR-Appserver/blob/31cddbfed0e6b3f3fca66d8dab9bef0a50ea443d/src/main/java/org/radarbase/appserver/service/protocol/GithubClient.java#L90
Or should it be added in addition?

Reply:
ah ok cool, I thought it is like the okhttp builder. Would have been nice to have all configs in the same place but ok if not possible.